File: algorithms/2_medium/G/00775_global_and_local_inversions/solution.c
static inline int abs_int(const int a)
{
    return a < 0 ? -a : a;
}

bool isIdealPermutation(const int* nums, const int len)
{
    assert(len >= 1 && nums != NULL);
    for (int i = 0; i < len; i += 1) {
        if (abs_int(nums[i] - i) > 1) {
            return false;
        }
    }

    return true;
}

File: algorithms/2_medium/G/00775_global_and_local_inversions/solution.rs
impl Solution {
    pub fn is_ideal_permutation(nums: Vec<i32>) -> bool {
        let len = nums.len();

        for i in 0..len {
            if (nums[i] - i as i32).abs() > 1 {
                return false;
            }
        }

        true
    }
}
